### DISCUSSION TITLE
How to open a Page in a new Window while maintaining state.

### DISCUSSION THREAD

**Discussion by technelak** [microsoft/microsoft-ui-xaml#8333] [Q&A]
Hello,

What is the correct way to open a Page in a new Window without losing UI state? I have tried several approaches, but I am unsure if I am on the right track. I attempted to use a new Frame to navigate to the Page and set this.NavigationCacheMode = NavigationCacheMode.Required;, but this does not seem to work. Additionally, I do not need a navigation state for this page.

I am struggling to figure out the correct way to copy the content of the page and set a new Frame's content to the content of the original page without getting a "Value does not fall within the expected range" error. Is it necessary to implement a service to save the state of each UI element on the page and reapply the state when the page loads again, or is there a simpler way to accomplish this? Thanks so much!

**Comment by ReginaldTang** [comment]
I think you need something to save the state of the page

**Comment by nikolayvpavlov** [comment]
You do not store you state in the page.  You store in a view-model, which you bind your page to.
